### ISSUE TITLE
Cypress 12.7.0 Pararell job pass properly, but it's not terminated that leads to timout in Gitlab CI 

### ISSUE THREAD

**Issue by Kozioleczek** [cypress-io/cypress#26062]
### Current behavior

We have been using Cypress in our project for more than a year. Since migrating Cypress to version 12.7.0, we have noticed that randomly our parallel tests end up with Gitlab CI being terminated by timeouts.

Based on the logs, we see that all the tests within the parallel test ended correctly. It's worth to mention that all specs pass properly locally.

Our test setup coincides almost 1:1 with the official Cypress documentation. 

### Desired behavior

We expect that if the job of the parallel test is successful, the job itself will end and this will happen every time.

### Test code to reproduce

Gitlab Job

```yaml
e2e:
  stage: E2E
  image: cypress/included:12.7.0
  timeout: 30 minutes
  parallel: 8
  script:
    - yarn start-server-and-test start:app http://localhost:3000 start:e2e # start:app runs our application, start:e2e 
```

Command `start:e2e` is declared in `package.json` as  `"start:e2e": npx cypress run --record --key $CYPRESS_DASHBOARD_KEY --headless --parallel --browser chrome`

### Cypress Version

12.7.0

### Node version

16.18.1

### Operating System

gitlab-runner 15.6.1

### Debug Logs

```shell
✔  All specs passed!                        08:18      109      109        -        -        -  
───────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                       
  Recorded Run: https://cloud.cypress.io/projects/###
Done in 631.29s.
ERROR: Job failed: execution took longer than 30m0s seconds
```

**Comment by RaceWalker** [comment]
I have a similar issue when performing jobs in parallel in the GitLab CI job freezes after Cypress execution. It seems like the Cypress process can not finish executing

**Comment by nagash77** [comment]
Hi @Kozioleczek , Are you using Cypress Cloud for parallelization?

**Comment by Kozioleczek** [comment]
@nagash77 Yeah I do

**Comment by bobbhatti** [comment]
Hi @nagash77 

I have experienced similar issues using Cypress 12.1.0 running in parallel using Cypress Cloud and Giltab CI - albeit we are using a self-hosted Gitlab Enterprise.

On numerous occasions, the majority of my tests will pass and complete, but one of the specs just carries on as if it's still executing and will eventually timeout.

I set the parallel instances to 3, running in a docker container (debian) against chromium headless.

However, am using cypress-cucumber-preprocessor plugin too.

Rgds
Bob

**Comment by nagash77** [comment]
@Kozioleczek Can you please open an issue in our [paid support portal](https://www.cypress.io/support/)?  That will get you the fastest support for Cloud related issue.

**Comment by nagash77** [comment]
Hi @bobbhatti ,  If you have a reproducible example please open a separate ticket and the team will look into it.

**Comment by nagash77** [comment]
I am closing this issue due to lack of activity until we get a reproducible example.
